src/blocked.c

```diff
@@ -101,14 +101,22 @@ void blockClient(client *c, int btype) {
  * he will attempt to reprocess the command which will update the statistics.
  * However in case the client was timed out or in case of module blocked client is being unblocked
  * the command will not be reprocessed and we need to make stats update.
- * This function will make updates to the commandstats, slowlog and monitors.*/
-void updateStatsOnUnblock(client *c, long blocked_us, long reply_us, int had_errors) {
+ * This function will make updates to the commandstats, slowlog and monitors.
+ * The failed_or_rejected parameter is an indication that the blocked command was either failed internally or
+ * rejected/aborted externally. In case the command was rejected the value ERROR_COMMAND_REJECTED should be passed.
+ * In case the command failed internally, ERROR_COMMAND_FAILED should be passed.
+ * A value of zero indicate no error was reported after the command was unblocked  */
+void updateStatsOnUnblock(client *c, long blocked_us, long reply_us, int failed_or_rejected) {
     const ustime_t total_cmd_duration = c->duration + blocked_us + reply_us;
     c->lastcmd->microseconds += total_cmd_duration;
     c->lastcmd->calls++;
     c->commands_processed++;
     server.stat_numcommands++;
-    if (had_errors) c->lastcmd->failed_calls++;
+    debugServerAssertWithInfo(c, NULL, failed_or_rejected >= 0 && failed_or_rejected <= ERROR_COMMAND_FAILED);
+    if (failed_or_rejected & ERROR_COMMAND_FAILED)
+        c->lastcmd->failed_calls++;
+    else if (failed_or_rejected & ERROR_COMMAND_REJECTED)
+        c->lastcmd->rejected_calls++;
```

Comment:
Shouldn't we just crash on the else block?

```suggestion
    if (failed_or_rejected & ERROR_COMMAND_FAILED)
        c->lastcmd->failed_calls++;
    else if (failed_or_rejected & ERROR_COMMAND_REJECTED)
        c->lastcmd->rejected_calls++;
    else
        debugServerAssertWithInfo(c, NULL, 0);
```

Reply:
Probably good idea. 10x!